lib/em-http/http_connection.rb

```diff
@@ -68,7 +67,8 @@ def ssl_handshake_completed
         return true
       end
 
-      unless OpenSSL::SSL.verify_certificate_identity(@last_seen_cert, host)
+      unless certificate_store.verify(@last_seen_cert) &&
```

Comment:
Now that you’re doing the verification at the end, there’s no need to call `#verify` in each `#ssl_verify_peer` call, I think.

I’ve implemented a similar approach in https://github.com/ably-forks/em-http-request/pull/2, in which I also stopped adding the certificates passed to `ssl_verify_peer` to the store, instead passing them to `#verify` as intermediate certificates. My understanding from the OpenSSL docs is that the store is intended as a store of trust anchors (it might be that there’s no problem with putting other stuff in there too, I don’t know, but I felt safer using it as the docs signalled it’s intended to be used).